exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/transactions/ValidThrowingTx.java

```diff
@@ -16,42 +16,38 @@
 
 package com.exonum.binding.qaservice.transactions;
 
-import static com.exonum.binding.qaservice.transactions.QaTransactionTemplate.newQaTransactionBuilder;
+import static com.exonum.binding.common.serialization.StandardSerializers.protobuf;
 import static com.exonum.binding.qaservice.transactions.TransactionPreconditions.checkTransaction;
 
-import com.exonum.binding.common.message.BinaryMessage;
-import com.exonum.binding.common.message.Message;
+import com.exonum.binding.common.serialization.Serializer;
 import com.exonum.binding.qaservice.QaSchema;
+import com.exonum.binding.qaservice.QaService;
 import com.exonum.binding.qaservice.transactions.TxMessageProtos.ValidThrowingTxBody;
-import com.exonum.binding.storage.database.Fork;
+import com.exonum.binding.transaction.RawTransaction;
 import com.exonum.binding.transaction.Transaction;
-import com.google.common.annotations.VisibleForTesting;
-import com.google.protobuf.InvalidProtocolBufferException;
+import com.exonum.binding.transaction.TransactionContext;
 import java.util.Objects;
 
 public final class ValidThrowingTx implements Transaction {
```

Comment:
I think we must drop `Valid` prefix as a transaction can be 'invalid' only if
its message has incorrect signature (and that is handled by the core).

Its message might also be invalid in the sense that it carries some non-decodable garbage, but, again, that can be tested in _0.10_ through the core endpoint.

**cc** @dip56